errors/chain.go

```diff
@@ -1,7 +1,7 @@
 package errors
 
-// ErrorChain is a linked list of error pointers that point to a parent above and a child below.
-type ErrorChain struct {
+// ChainError is a linked list of error pointers that point to a parent above and a child below.
```

Comment:
This is not backwards compatible and it is exported.

Reply:
yes so this will be v4 change

Reply:
You can just set a type alias with `type ErrorChain = ChainError` and avoid that issue.  Is there something wrong with the original name?

Reply:
linting error there is a specific syntax. so i think if i try to type alias it might still complain. So moving forward with proper naming convention and doing a major release is better

Reply:
The linter is just a suggestion, but in this case there is nothing wrong with the type name -- it is clear that it isn't meant to be an actual error object if that is what the linter is complaining about.  Also I would never regard a linter's suggestion as more important than maintaining backwards compatibility unless it were a major security issue or an outright bug.

Reply:
an example from uber-go guide for customer error

https://github.com/uber-go/guide/blob/master/style.md#errors

For custom error types, use the suffix Error instead.

```
// Similarly, this error is exported
// so that users of this package can match it
// with errors.As.

type NotFoundError struct {
  File string
}

func (e *NotFoundError) Error() string {
  return fmt.Sprintf("file %q not found", e.File)
}

// And this error is not exported because
// we don't want to make it part of the public API.
// We can still use it inside the package
// with errors.As.

type resolveError struct {
  Path string
}

func (e *resolveError) Error() string {
  return fmt.Sprintf("resolve %q", e.Path)
}
```

Reply:
details from golang wiki https://github.com/golang/go/wiki/Errors#naming

Reply:
done